python/ray/data/tests/test_optimize.py

```diff
@@ -391,289 +316,6 @@ def test_write_doesnt_reorder_randomize_block(ray_start_regular_shared, tmp_path
     assert "Write" in stats, stats
 
 
-def test_optimize_fuse(ray_start_regular_shared):
```

Comment:
for the test cases in this file, it looks like there's no test coverage for `Dataset`, after we remove the tests for `DatasetPipeline`. Can you port the unit tests to `Dataset` if feasible?

Reply:
These tests check stage fusion. Since we're removing stages, I think it should be okay to remove these tests (and the corresponding configurations like `optimize_fuse_read_stages`)? 

Also, most of these configurations don't work with `Dataset`. For example:

```python
import ray

ray.data.DataContext.get_current().optimize_fuse_read_stages = False

ds = ray.data.range(1, parallelism=1).map_batches(lambda x: x)
ds.take()
# Read operator gets fused despite setting `optimize_fuse_read_stages=False`
# InputDataBuffer[Input] -> TaskPoolMapOperator[ReadRange->MapBatches(<lambda>)] -> LimitOperator[limit=20]
```

Reply:
yeah, i think the operator fusion logic in the new plan optimizer should be covered under `test_execution_optimizer.py` now. https://github.com/ray-project/ray/blob/master/python/ray/data/tests/test_execution_optimizer.py

Reply:
Ok, if it's covered by other tests, it's okay to remove here.